tests/unit/test_runtime.py

```diff
@@ -967,6 +967,72 @@ async def test_ipython_agentskills_fileop_pwd(temp_dir, box_class, enable_auto_l
     await asyncio.sleep(1)
 
 
+@pytest.mark.asyncio
+async def test_ipython_agentskills_fileop_pwd_with_userdir(temp_dir, box_class):
```

Comment:
I am not sure but looks like this test doesn't validate this PR (i.e. keep prompt or not)?

Reply:
This primarily tests for another issue (e.g., the IPython workspace will broken if agent cd to `~` in command line. I just added a test specifically for this keep_prompt=True